api/src/main/java/eu/darkbot/api/extensions/MapGraphics.java

```diff
@@ -0,0 +1,296 @@
+            drawPoly(type, positions.toArray(new Point[0]));
+        else if (positions.iterator().next() instanceof Locatable)
+            drawPoly(type, positions.toArray(new Locatable[0]));
+    }
+
+    /**
+     * Draws a line
+     *
+     * @param x1 the first point's <i>x</i> coordinate.
+     * @param y1 the first point's <i>y</i> coordinate.
+     * @param x2 the second point's <i>x</i> coordinate.
+     * @param y2 the second point's <i>y</i> coordinate.
+     */
+    default void drawLine(int x1, int y1, int x2, int y2) {
+        getGraphics2D().drawLine(x1, y1, x2, y2);
+    }
+
+    /**
+     * Draws a line
+     *
+     * @param a start
+     * @param b end
+     */
+    default void drawLine(Point a, Point b) {
+        drawLine(a.x(), a.y(), b.x(), b.y());
+    }
+
+    /**
+     * Draws a line
+     *
+     * @param a start
+     * @param b end
+     */
+    default void drawLine(Locatable a, Locatable b) {
+        drawLine(toScreenPoint(a), toScreenPoint(b));
+    }
+
+    /**
+     * @return string width using current font
+     */
+    default int getStringWidth(String str) {
+        return getGraphics2D().getFontMetrics().stringWidth(str);
+    }
+
+    /**
+     * Draws a String at given position
+     *
+     * @param str         String to draw
+     * @param loc         where string will be drawn
+     * @param heightAlign height align added after translate
+     * @param stringAlign String align
+     */
+    default void drawString(Locatable loc, String str, int heightAlign, StringAlign stringAlign) {
+        drawString(toScreenPointX(loc.getX()), toScreenPointY(loc.getY()) + heightAlign, str, stringAlign);
+    }
+
+    /**
+     * Draws a String at given position
+     *
+     * @param str         String to draw
+     * @param point       where string will be drawn
+     * @param stringAlign String align
+     */
+    default void drawString(Point point, String str, StringAlign stringAlign) {
+        drawString(point.x(), point.y(), str, stringAlign);
+    }
+
+    default void drawString(int x, int y, String str, StringAlign stringAlign) {
+        if (str == null || str.isEmpty()) return;
+
+        if (stringAlign != StringAlign.LEFT)
+            x -= getStringWidth(str) >> (stringAlign == StringAlign.MID ? 1 : 0);
+
+        getGraphics2D().drawString(str, x, y);
+    }
+
+    default void drawBackgroundedText(int x, int y, String str, Color backgroundColor, StringAlign stringAlign) {
+        if (str == null || str.isEmpty()) return;
+
+        FontMetrics fontMetrics = getGraphics2D().getFontMetrics();
+
+        int strWidth = getStringWidth(str);
+        int strHeight = fontMetrics.getMaxAscent();
+
+        if (stringAlign != StringAlign.LEFT)
+            x -= strWidth >> (stringAlign == StringAlign.MID ? 1 : 0);
+
+        Color textColor = getGraphics2D().getColor();
+
+        setColor(backgroundColor);
+        drawRect(x, y - strHeight, strWidth, strHeight + fontMetrics.getMaxDescent(), true);
```

Comment:
If the draw point is y - strHeight, it means we're anchoring on the baseline. This may, or may not be intended, as by default drawing strings is anchored to top-right, and with strirng-align one may expect to only move sideways (anchor  right, left, or mid).

Maybe we need string alignment to be more than an enum of 3 values, maybe:
`Anchor.of(TOP, MIDDLE)`
where one is vertical anchor point, and the other is horizontal, a bit like css vertical-align (baseline, top, bottom, middle) and text-align (left, right, center)